Bottom: a7a4e2898b5cea41e3f03bc78fc9e4f2504e2659
Top:    97f1212a1e259cfe5fe72aa256376091084c1b04
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-13 02:54:58 +0000

Refresh of draw-polygons-in-a-more-intell

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index fb6a054..115b353 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -914,7 +914,8 @@ hidgl_fill_pcb_polygon (PolygonType *poly, const BoxType *clip_box)
   if (poly->Clipped == NULL)
     return;
 
-  use_new_stencil = polygon_contains_user_holes (poly);
+  use_new_stencil = polygon_contains_user_holes (poly) ||
+                    TEST_FLAG (FULLPOLYFLAG, poly);
 
   fill_polyarea (poly->Clipped, clip_box, use_new_stencil);
